pkg/chunk/grpc/index_client.go

```diff
@@ -59,8 +59,10 @@ func (s *StorageClient) query(ctx context.Context, query chunk.IndexQuery, callb
 		if err == io.EOF {
 			break
 		}
-		if !callback(readBatch) {
-			return nil
+		if readBatch != nil {
```

Comment:
We should check the error here tbh.

Reply:
What do you mean @gouthamve? The error is checked above. However, I'm not sure why we need this `if` at all.

Reply:
@pracucci Though we check the error. If the response is nil the callback func fails. I am just trying to make sure if grpc server doesn't handle all cases. cortex shouldn't crash in any scenario.

Reply:
If the gRPC server is buggy and returns `nil` batch and `nil` error (which is a case we never expect to happen) you will end up silently ignoring it and you would potentially end up with wrong query results. If you would have the same exact bug in Cortex today, Cortex would panic. I don't see why we should do it differently with the gRPC store.

Reply:
Okay. In that case, I can remove the additional ```nil``` value validation.